tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlockAwareSegmentInputStreamImpl.java

```diff
@@ -161,6 +211,45 @@ private List<ByteBuf> readNextEntriesFromLedger(long start, long maxNumberEntrie
         }
     }
 
+    @Override
+    public int read(byte[] b, int off, int len) throws IOException {
+        if (b == null) {
+            throw new NullPointerException();
```

Comment:
IllegalArgumentExcepton? otherwise can as well skip this and let the code below throw NPE.

Reply:
I think throw NPE is ok as well? I add information in the exception.